env/fs_posix.cc

```diff
@@ -325,14 +327,16 @@ class PosixFileSystem : public FileSystem {
     SetFD_CLOEXEC(fd, &options);
 
     if (options.use_mmap_writes) {
-      if (!checkedDiskForMmap_) {
-        // this will be executed once in the program's lifetime.
-        // do not use mmapWrite on non ext-3/xfs/tmpfs systems.
-        if (!SupportsFastAllocate(fname)) {
-          forceMmapOff_ = true;
+      std::call_once(check_disk_for_mmap_once_, [&fname, this]() {
```

Comment:
I would be ok with this as a "better than before" fix if the two occurrences of this code were consolidated into one function and the `checkedDiskForMmap_` field was removed (because of just one `call_once`)

Reply:
On second thought, I plan to pursue this direction without changing behavior in a PR dedicated to fixing TSAN-error. We can follow up with potential fixes/optimizations in future PRs @pdillinger @mrambacher